packages/loot-core/src/server/budget/actions.js

```diff
@@ -272,3 +284,434 @@ export async function setCategoryCarryover({ startMonth, category, flag }) {
     }
   });
 }
+
+export async function applyTemplate({ month }) {
+  await processTemplate(month, false);
+}
+
+export async function overwriteTemplate({ month }) {
+  await processTemplate(month, true);
+}
+
+async function processTemplate(month, force) {
+  let category_templates = await getCategoryTemplates();
+
+  let categories = await db.all(
+    'SELECT * FROM v_categories WHERE tombstone = 0'
+  );
+
+  let num_applied = 0;
+  for (let c = 0; c < categories.length; c++) {
+    let category = categories[c];
+
+    let budgeted = await getSheetValue(
+      monthUtils.sheetForMonth(month),
+      `budget-${category.id}`
+    );
+
+    if (budgeted === 0 || force) {
+      let template = category_templates[category.id];
+
+      if (template) {
+        let to_budget = await applyCategoryTemplate(
+          category,
+          template,
+          month,
+          force
+        );
+        if (to_budget != null) {
+          num_applied++;
+          await setBudget({ category: category.id, month, amount: to_budget });
+        }
+      }
+    }
+  }
+  if (num_applied === 0) {
+    console.log('All categories were up to date.');
+  } else {
+    console.log(`${num_applied} categories updated.`);
+  }
+}
+
+async function getCategoryTemplates() {
+  const matches = [
+    {
+      type: 'simple',
+      re: /^#template \$?(\-?\d+(\.\d{2})?)$/im,//eslint-disable-line
```

Comment:
Is there a way to define these in a more structured way? (or at least remove the `#template` part?)

Reply:
I think the reason he wrote it this way is because the 'notes' are used to enter the string.  The `#template`, I think, helps keep false positives from being applied and still allows the 'notes' to be used for actual notes.  This is not something I wrote and am not really familiar with, so if you would like to change it, please feel free.  A different tag could be appropriate instead of `#template` to make it obvious this is a goal string.